src/hdmf/common/table.py

```diff
@@ -731,25 +761,40 @@ def add_column(self, **kwargs):  # noqa: C901
             if isinstance(index, VectorIndex):
                 col_index = index
                 self.__add_column_index_helper(col_index)
-            elif isinstance(index, bool):  # make empty VectorIndex
-                if len(col) > 0:
-                    raise ValueError("cannot pass empty index with non-empty data to index")
-                col_index = VectorIndex(name + "_index", list(), col)
+            elif isinstance(index, bool):
```

Comment:
This code might be simplified by saying earlier:
```
if index is True:
    index = 1
```
and removing the `isinstance(index, bool)` checks